lib/Importer/ProtobufLoader.cpp

```diff
@@ -34,6 +34,22 @@ llvm::Expected<Tensor *> ProtobufLoader::getTensorByName(llvm::StringRef name) {
   return tensors_[name].get();
 }
 
+llvm::Expected<std::unique_ptr<Tensor>>
+ProtobufLoader::takeTensorByName(llvm::StringRef name) {
+  auto it = tensors_.find(name);
+  RETURN_ERR_IF_NOT(
+      it != tensors_.end(),
+      llvm::Twine("There is no tensor registered with name ", name).str());
+
+  std::unique_ptr<Tensor> t = std::move(it->second);
+
+  RETURN_ERR_IF_NOT(
```

Comment:
getTensorByName now will return nullptr if we take it here first (probably not expected), maybe add similar condition to the getTensorByName.